### PR TITLE
[IEEE-245]: Get Review Status By Username

### PR DESCRIPTION
**Pull request by liu-samuel**
## Overview

- Resolves IEEE-245
- Added a new endpoint to get a user's review status based on username

## Unit Tests Created

- User Review Status Unit Tests in test_api.py

## Steps to QA

- N/A

### CODE REVIEW

**Review by LunarFang416 — DISMISSED** [review]
LGTM

File: hackathon_site/event/api_views.py [hunk 1]
```diff
@@ -62,6 +62,24 @@ def get(self, request, *args, **kwargs):
         return self.retrieve(request, *args, **kwargs)
 
 
+class UserReviewStatusAPIView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    """
+    View to handle review status status of a user with a certain username
```
Comment (lukecheseldine) [inline hackathon_site/event/api_views.py]:
```suggestion
    View to handle review status of a user with a certain username
```

**Review by lukecheseldine — CHANGES_REQUESTED** [review]
great work Sam! this is almost ready, I'd just like you to add two more test cases.
1. the api is hit with an email that doesn't exist (return 400)
2. the api is hit without being logged in (return 401)
in general, we should try to write tests for all possible return codes. write these up and should be good to go!

File: hackathon_site/event/api_views.py (hunk 1 above)
Comment (liu-samuel) [reply hackathon_site/event/api_views.py]: Changes are pushed

**Review by lukecheseldine — APPROVED** [review]
looks good, thanks for making those changes :)

**Review by LunarFang416 — APPROVED** [review]
👍👍

**Pull request merged**